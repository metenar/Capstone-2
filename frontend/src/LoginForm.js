import React from "react";
import {useHistory} from "react-router-dom";
import 'bootstrap/dist/css/bootstrap.css';
import { Form, FormGroup, Label, Input} from "reactstrap";
import {Button} from "reactstrap"
import { useFormik } from 'formik';

const validate = values => {
    const errors = {};
    if (!values.username) {
      errors.username = 'Required';
    } else if (values.username.length > 15) {
        errors.username = 'Must be 15 characters or less';
    }
  
    if (!values.password) {
      errors.password = 'Required';
    } else if (values.password.length < 6) {
        errors.password = 'Must be at least 6 characters';
    }
  
    return errors;
  };


const LoginForm=({login})=>{
    const history = useHistory();
    const formik = useFormik({
      initialValues : {
        username : '',
        password : ''
      },
      validate,
      onSubmit      : async (values) => {
        try {
          let res=await login(values);
          if(res.success){
            history.push('/')
        }
        } catch (err) {
          formik.errors.password = err;
        }
      }
    });
    
    
    return (
        <section className="col-md-4 m-auto">
        <Form onSubmit={formik.handleSubmit}>
            <h2>Login</h2>
            <FormGroup>
                <Label for="username">Username </Label>
                <Input 
                    type="text"
                    id="username"
                    name="username"
                    value={formik.values.username}
                    placeholder="Enter username"
                    onChange={formik.handleChange}
                    onBlur={formik.handleBlur}
                />
                {formik.touched.username && formik.errors.username ? <div className="alert alert-danger">{formik.errors.username}</div>:null}
            </FormGroup>
            <FormGroup>
                <Label for="password">Password </Label>
                <Input 
                    type="password"
                    id="password"
                    name="password"
                    value={formik.values.password}
                    placeholder="Enter your password"
                    onChange={formik.handleChange}
                    onBlur={formik.handleBlur}
                />
                {formik.touched.password && formik.errors.password ? <div className="alert alert-danger">{formik.errors.password}</div>:null}
            </FormGroup>
            <Button color="primary">Submit</Button>
        </Form>
        </section>
    )
}
export default LoginForm;