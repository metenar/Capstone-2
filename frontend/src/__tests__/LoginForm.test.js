import { render,screen} from '@testing-library/react';
import { MemoryRouter, Route, Switch} from "react-router-dom";
import LoginForm from '../LoginForm';
import '@testing-library/jest-dom'
import CurrentUserContext from "../CurrentUserContext"

function renderLogin(currentUser) {

    return render(
        <MemoryRouter initialEntries={["/login"]}>
            <CurrentUserContext.Provider value={currentUser}>
                <Switch>
                    <Route exect path="/login">
                        <LoginForm />
                    </Route>
                </Switch>
            </CurrentUserContext.Provider>
        </MemoryRouter>
    );
  }

  const currentUser=[
    {
      username:"metenar",
      firstName:"Mete",
      lastName:"NAR",
      email:'metenar@gmail.com',
      password:"",
      image:"www.mete.jpg"
    }]

test('renders without crushing', () => {
  renderLogin({currentUser});
});

test('should match the snapshot', () => {
  const {asFragment}=renderLogin({currentUser});
  expect(asFragment()).toMatchSnapshot()
});

test('should Login visible', ()=>{
    renderLogin({currentUser});
    expect(screen.queryAllByText('Login')).not.toBeNull();
})