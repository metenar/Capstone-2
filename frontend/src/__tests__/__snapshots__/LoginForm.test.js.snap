// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4 m-auto"
  >
    <form
      class=""
    >
      <h2>
        Login
      </h2>
      <div
        class="form-group"
      >
        <label
          class=""
          for="username"
        >
          Username 
        </label>
        <input
          class="form-control"
          id="username"
          name="username"
          placeholder="Enter username"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class=""
          for="password"
        >
          Password 
        </label>
        <input
          class="form-control"
          id="password"
          name="password"
          placeholder="Enter your password"
          type="password"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
